src/rust/engine/fs/store/src/remote.rs

```diff
@@ -122,7 +151,7 @@ impl ByteStore {
     // could be mutated by another process. We guard against this by creating an anonymous
     // temporary file and ensuring it is written to and closed via the only other handle to it in
     // the code just above.
-    let mmap = unsafe {
+    let mmap = Arc::new(unsafe {
```

Comment:
Using an Arc here should be safe because this memory is mapped immutably and, in any event, it should only be accessed by one async task at a time.